### ISSUE TITLE
Backwards engineering the driving video from the generated video

### ISSUE THREAD

**Issue by Bardo-Konrad** [AliaksandrSiarohin/first-order-model#184]
Having created some deepfakes, I started to wonder how you can extract the driving video from the generated video without additional information.

As rotation is possible, the first-order-model must create an internal 3D structure of the face. On the other hand proportions are being aligned, distances are being fitted.

Therefore I would say: not really.

You might recognize tics the actor in the driving video has, which might give you a hint who he is, but not based on the video alone. Especially without any other information.

In short: is this a destructive process?

What are your thoughts on this?

**Comment by AliaksandrSiarohin** [comment]
Did you mean absolute or relative animation?

**Comment by Bardo-Konrad** [comment]
Relative.

**Comment by Bardo-Konrad** [comment]
Still interested in your thoughts on this, @AliaksandrSiarohin

**Comment by AliaksandrSiarohin** [comment]
I guess you could try the following protocol:
1) Generate some synthetic images with corresponding 3d Deformable face model. (You can took an existing dataset).
2) Compute keypoints and jacobians for these images. 
3) Fit some model (neural net) to predict  parameters of this model from keypoints and jacobians.
4) Evaluate how good is the predicted face model.

**Comment by AliaksandrSiarohin** [comment]
This is for absolute animation, for relative you first need to restore absolute keypoints.

**Comment by Santiblocks** [comment]
I was also thinking about this! Definitely something to be aware of, considering all that nn's can do
I would expect this to be a destructive process tho
